protocol.md

```diff
@@ -1078,13 +1078,24 @@ interface Hover {
 	contents: MarkedString | MarkedString[];
 	
 	/**
-	 * An optional range
+	 * An optional range is a range inside a text document 
+	 * that is used to visualize a hover, e.g. by changing the background color.
 	 */
 	range?: Range;
 }
 ```
 Where `MarkedString` is defined as follows:
 ```typescript
+/**
+ * The marked string is rendered:
+ * - as markdown if it is represented as a string
```

Comment:
I just state how they are handled now. Agreed about `usually` for markdown.

I don't see why `MarkedString` is better than `string`? I would prefer strings, since they are generic and can encode code blocks. It would simplify specification to a simple fact that `marked string are usually rendered as markdown`. Opposite is not true, one need to come up with a special handling of a language to allow arbitrary markdown.